src/pages/DistributePage/DistributePage.tsx

```diff
@@ -38,15 +47,32 @@ function DistributePage() {
   const [loadingTrx, setLoadingTrx] = useState(false);
   const [updateAmount, setUpdateAmount] = useState(0);
   const [selectedVaultId, setSelectedVaultId] = useState('');
+
   const contracts = useContracts();
   const currentOrg = useCurrentOrg();
-  const vaults = useVaults(currentOrg.data?.id);
+  const { isLoading: vaultLoading, data: vaults } = useVaults(
+    currentOrg.data?.id as number
+  );
+
   const { uploadEpochRoot } = useDistributor();
-  const [selectedVault, setSelectedVault] = useState<IVault | undefined>();
+  const [selectedVault, setSelectedVault] = useState<
+    GraphQLTypes['vaults'] | undefined
+  >();
   const { apeError } = useApeSnackbar();
+  const [distributionDTO, setDistributionDTO] =
+    useState<ValueTypes['distributions_insert_input']>();
+  const { mutateAsync } = useSaveEpochDistribution(distributionDTO);
```

Comment:
now that we're using `useMutation` from `react-query`, we can avoid this weirdness and just pass `distribution` into the mutateAsync as an ordinary argument, right?

Reply:
Done.